tests/data_transfer_tests/services/test_inventory.py

```diff
@@ -36,3 +36,34 @@ def test_normalise_day() -> None:
     result = one == two
 
     assert result
+
+
+def test_all_devices_by_type_cache_success() -> None:
+    inventory.all_devices_by_type.cache_clear()
+    num_requests = 10
+
+    with patch("requests.get", return_value=Mock()) as _:  # act
```

Comment:
RE: `how come you have used a fixture to mock out requests? I discovered it's possible to do something similar but to override "requests.get" directly`

In my case I need a specific return value if you call something specific. Could indeed specify the returned value but using a mocked session with adapters allows for dynamic responses (which I used for the ByteFlies - but might not be as needed here)

Reply:
You can set/obtain/change the response using [MagicMock](https://github.com/ideafast/middleware-services/blob/2e300907814126a968f1cfe344940a4924a7c541/tests/data_transfer_tests/services/test_inventory.py#L60-L61). So if I wanted, I could do `result = inventory.all_devices_by_type(utils.DeviceType.BTF)` and that would contain the result of `requests.get`. That way, you do not need sessions/adapters 🚀 

The reason I had to override `.json` (as a function using lambda) is because it gets called in `all_devices_by_type`